common/gateway-storage/migrations/20240910120000_generic_client_id.sql

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright 2024 - Nym Technologies SA <contact@nymtech.net>
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+CREATE TABLE clients (
+   id              INTEGER NOT NULL PRIMARY KEY AUTOINCREMENT,
+   client_type     TEXT NOT NULL CHECK(client_type IN ('entry mixnet', 'entry wireguard', 'exit wireguard'))
+);
+
+INSERT INTO clients (id, client_type)
+SELECT id, 'entry mixnet'
+FROM shared_keys;
+
+CREATE TABLE shared_keys_tmp (
+   client_id                                INTEGER NOT NULL PRIMARY KEY REFERENCES clients(id),
```

Comment:
I'm not entirely sure whether this creates the relation you're looking for.
since `shared_keys` references `clients`, you will end up having multiple of those for the same underlying user (but in different mode).

how about something like this:
- the "master" `client` table that has the id, address and shared key
- bandwidth table with extra `client_type` referencing that master client table

so that the same entity would end up having different bandwidth metering based on its type

Reply:
Yes, but what would the shared key and address be for the "master" client that is not of mixnet type then? This is exactly why I created the `clients` table.
Have multiple entries of in `clients` for the same underlying user doesn't seem like a problem to me, as each mode is essentially a different "client", the user doesn't have anything to do with it. The same user will actually have all 4 instances of client modes running at the same time.